packages/SwingSet/src/kernel/vatManager/deliver.js

```diff
@@ -116,15 +116,17 @@ export function makeDeliver(tools, dispatch) {
   }
 
   async function replayTranscript() {
-    transcriptManager.startReplay();
-    for (const t of vatKeeper.getTranscript()) {
+    if (!vatKeeper.isDead()) {
```

Comment:
This `replayTranscript` function is specific to the particular vatManager type, so it's not the best place to do this `isDead` check. Other (future) vatManager types, especially ones that use snapshots instead of transcripts, won't pass through here, and those need to be skipped too.

Instead, put the check in `kernel.js` at the beginning of the loop where we call `vat.manager.replayTranscript()`, so it looks like:

```js
    // if it *was* initialized, replay the transcripts
    if (wasInitialized) {
      console.info('Replaying SwingSet transcripts');
      const oldLength = kernelKeeper.getRunQueueLength();
      for (const vatID of ephemeral.vats.keys()) {
        const vat = ephemeral.vats.get(vatID);
        if (vat.dead) {
          console.debug(`Skipping reload of dead vat ${vatID}`);
          continue;
        }
        console.debug(`Replaying transcript of vatID ${vatID}`);
        // eslint-disable-next-line no-await-in-loop
        await vat.manager.replayTranscript();
        ...
```